# Data Validation And Drift

Check missingness, feature summaries, and population drift before modeling. This tests diagnostic notebooks that are not primarily training runs.

In [1]:
from pathlib import Path

exec(Path("samples/ml_tasks/data_validation_drift.py").read_text(encoding="utf-8"))


drift_artifact: artifacts/ml_tasks/data_validation_drift.json
features_checked: 5
alerts: 3
- usage: psi=0.251, missing=0, mean_shift=0.430
- support_load: psi=0.328, missing=14, mean_shift=0.415
- nps: psi=0.084, missing=19, mean_shift=-0.233


## Finding: validation and drift alerts

Clean execution completed successfully. The workflow checked 5 features and raised 3 alerts:

- `usage`: PSI 0.251, missing 0, mean shift +0.430
- `support_load`: PSI 0.328, missing 14, mean shift +0.415
- `nps`: PSI 0.084, missing 19, mean shift -0.233

Artifact produced: `artifacts/ml_tasks/data_validation_drift.json`.

The notebook tells a usable validation/drift story because it identifies the affected features, separates drift from missingness, and leaves a machine-readable report for downstream review.
